dyn_sim/sim/simulator.py

```diff
@@ -104,18 +104,33 @@ def animate(
         anim_name : str, default=None
             Name for animation file. If not None, saves an mp4. Else, doesn't save.
         """
-        xlim, ylim, zlim = lims
+        (
+            xlim,
+            ylim,
+            zlim,
+        ) = lims  # for 2D animations, still need to pass in a zlim (though it is now unused)
 
         self._fig = plt.figure()
-        self._ax = p3.Axes3D(self._fig)
-        self._ax.set_proj_type("ortho")
-        self._ax.grid(False)
-        self._ax.set_xticks([])
-        self._ax.set_yticks([])
-        self._ax.set_zticks([])
-        self._ax.set_xlim3d(xlim)
-        self._ax.set_ylim3d(ylim)
-        self._ax.set_zlim3d(zlim)
+        if self._sys._is3d:
+            self._ax = p3.Axes3D(self._fig)
+            self._ax.set_proj_type("ortho")
+            self._ax.grid(False)
+            self._ax.set_xticks([])
+            self._ax.set_yticks([])
+            self._ax.set_zticks([])
+            self._ax.set_xlim3d(xlim)
+            self._ax.set_ylim3d(ylim)
+            self._ax.set_zlim3d(zlim)
+        else:
+            self._ax = plt.Axes(
+                self._fig, [0, 0, self._fig.get_figwidth(), self._fig.get_figheight()]
```

Comment:
Axes() demands a rectangle of some kind in its constructor - but neither of these options here work, both resulting in blank white plots with no animation going on

Reply:
let's put a pin in it and add 2D animation in a different PR. i'll work on it after we merge segway with the mpc stuff.